stdlib/camlinternalLazy.ml

```diff
@@ -41,9 +41,7 @@ let force_val_lazy_block (blk : 'arg lazy_t) =
   let closure = (Obj.obj (Obj.field (Obj.repr blk) 0) : unit -> 'arg) in
   Obj.set_field (Obj.repr blk) 0 raise_undefined;
   let result = closure () in
-  (* do set_field BEFORE set_tag *)
```

Comment:
What were these comments for? Thread safety? (Otherwise, should they be reproduced in the C implementation?)

Reply:
The GC must not be allowed to observe `Forward_tag` before the field has been updated. This potentially mattered when the sequence was in OCaml (it's hard to predict when the GC will run), but the order does not matter in C (the sequence cannot invoke the GC, so nothing can observe the order).